standards/security_standards.md

```diff
@@ -0,0 +1,440 @@
+
+3.9.2. TLS certificates must be valid and have the correct domain name, not be expired, and be installed with intermediate certificates when required.
+
+3.9.3. Failed TLS connections must not fall back to an insecure connection.
+
+3.9.4. Utilize TLS connections for all content requiring authenticated access and for all other sensitive information.
+
+3.9.5. Utilize TLS for connections to external systems that involve sensitive information or functions.
+
+3.9.6. Do Not Mix TLS and Non-TLS Content.
+
+3.9.7. Use the "Secure" Cookie Flag.
+
+3.9.8. Prevent Caching of Sensitive Data.
+
+3.9.9. Use HTTP Strict Transport Security.
+
+3.10. Malicious Code
+
+3.10.1. Application source code and third party libraries must not contain unauthorized phone home or data collection capabilities.
+
+3.10.2. Application must not ask for unnecessary or excessive permissions to privacy related features or sensors, such as contacts, cameras, microphones, or location.
+
+3.10.3. Application source code and third party libraries must not contain back doors, such as hard-coded or additional undocumented accounts or keys, code obfuscation, undocumented binary blobs, rootkits, or anti-debugging, insecure debugging features, or otherwise out of date, insecure, or hidden functionality that could be used maliciously if discovered.
+
+3.10.4. Application must employ integrity protections, such as code signing or subresource integrity. The application must not load or execute code from untrusted sources, such as loading includes, modules, plugins, code, or libraries from untrusted sources or the Internet.
+
+3.10.5. Application must have protection from subdomain takeovers if the application relies upon DNS entries or DNS subdomains, such as expired domain names, out of date DNS pointers or CNAMEs, expired projects at public source code repos, or transient cloud APIs, serverless functions, or storage buckets (autogen-bucket- id.cloud.example.com) or similar.
+
+3.10.6. Protections can include ensuring that DNS names used by applications are regularly checked for expiry or change.
+
+3.11. Files and Resources
+
+Application must satisfies the following high level requirements:
+
+- Untrusted file data must be handled accordingly and in a secure manner.
+- Untrusted file data obtained from untrusted sources must be stored outside the web root and with limited permissions.
+
+3.11.1. Specify list of allowed extensions. Only allow safe and critical extensions for business functionality. Ensure that input validation is applied before validating the extensions.
+
+3.11.2. Validate the file type, don't trust the Content-Type header as it can be spoofed.
+
+3.11.3. Change the filename to something generated by the application.
+
+3.11.4. Set a filename length limit. Restrict the allowed characters if possible.
+
+3.11.5. Set a file size limit.
+
+3.11.6. Only allow authorized users to upload and access uploaded files.
```

Comment:
Access is case-to-case basis - maybe let's reword it to "access authenticated uploaded files"?

Reply:
I re-read it now and yeah it makes sense.